### PR TITLE
doc+example changelog

### LINKED ISSUE #856: [ng] Changelog for documentation and examples?
Hi,

I had really hard times with porting my older Angular code today. And the reason is that there are big changes in the latest versions of Angular but there is no information about all of them (I know that there is this: https://pub.dartlang.org/packages/angular2#pub-pkg-tab-changelog but not everything is there - and specifically things I had to solve).

What about having some page where will be written all changes you have done in the source code of examples? I think this would help a lot with porting also our code.

My problem was the change in directives (e.g. that the constants need to be explicitly mentioned in components now) but this change was not written anywhere. And also the change in imports from angular2/core.dart to angular2/angular2.dart.

I dont have time to read Gitter...

### PR DESCRIPTION
**Pull request by kwalrath**
Staged at: https://kw-webdev-dartlang-1.firebaseapp.com/angular/version#!#changelog

/cc @matanlurey @alorenzen @Janamou 

Fixes #856

### CODE REVIEW

**Comment by chalin** [comment]
This will be a very good addition!

Before I comment about the content, I'd like to suggest an organizational change:

- IMHO rather than Changelog, this should be called an "Abridged Migration Guide" (or maybe "Migration Guide (Abridged)"), because the purpose is to "help you update your own code" (as it says in the intro), not to record changes to the docs.
- This migration guide should be on its own page so that we can benefit from the autogenerated ToC, which should contain entries for: AngularDart 4.0.0-alpha, Dart 1.24, AngularDart 3.1, etc.

WDYT?

**Comment by kwalrath** [comment]
> IMHO rather than Changelog, this should be called an "Abridged Migration Guide" (or maybe "Migration Guide (Abridged)"), because the purpose is to "help you update your own code" (as it says in the intro), not to record changes to the docs.

I guess I don't want to overpromise. We'll only be recording changes we've made (although I'll add FORM_DIRECTIVES anyway), and we can use this as a way of letting people know about substantive doc updates. I wouldn't mind it becoming its own page eventually, but this really is more of a changelog than a migration guide. I don't mind taking out the purpose, since that's just the impetus for creating the content, not necessarily what the content will be.

**Comment by chalin** [comment]
Ok.

File: src/angular/version.md [hunk 1]
```diff
@@ -117,3 +117,132 @@ which is documented at [angular.io](https://angular.io).
...
+others are the result of new recommendations or documentation.
+
+To see all changes to the docs and examples,
+go to the site-webdev repo on GitHub:
+
+* [Documentation changes](https://github.com/dart-lang/site-webdev/commits/master/src/angular)
+* [Example changes](https://github.com/dart-lang/site-webdev/commits/master/examples/ng/doc)
+
+
+### AngularDart 4.0 (alpha)
```
Comment (chalin) [inline src/angular/version.md]: I would like to see all of these headings appear in the autogenerated ToC. How about changing them to level 2s?

File: src/angular/version.md [hunk 2]
```diff
@@ -117,3 +117,132 @@ which is documented at [angular.io](https://angular.io).
...
+This summary of changes to the AngularDart documentation and examples
+can help you update your own code.
+Many of these changes are due to new releases of
+AngularDart, AngularDart Components, or the Dart SDK;
+others are the result of new recommendations or documentation.
+
+To see all changes to the docs and examples,
+go to the site-webdev repo on GitHub:
+
+* [Documentation changes](https://github.com/dart-lang/site-webdev/commits/master/src/angular)
```
Comment (chalin) [inline src/angular/version.md]: Maybe have this reference `.../src`.

File: src/angular/version.md [hunk 3]
```diff
@@ -117,3 +117,132 @@ which is documented at [angular.io](https://angular.io).
...
+{% endcomment %}
+* Changed API doc URLs
+  * The angular2 &rarr; angular change affected API doc URLs. <br>
+    Example:
+    [.../angular2/NgFor-class](https://webdev.dartlang.org/angular/api/angular2/NgFor-class)
+    &rarr;
+    [.../angular/NgFor-class](https://webdev-dartlang-org-dev.firebaseapp.com/angular/api/angular/NgFor-class)
+
+
+The removal of `angular2/common.dart` didn't affect our examples,
```
Comment (chalin) [inline src/angular/version.md]: It did affect our examples, its just that we made the change earlier, but it is actually a 4.x change.

File: src/angular/version.md [hunk 4]
```diff
@@ -117,3 +117,132 @@ which is documented at [angular.io](https://angular.io).
...
+
+
+More information:
+
+* [angular changelog](https://pub.dartlang.org/packages/angular#pub-pkg-tab-changelog)
+* [History for site-webdev/examples/ng (4-dev branch)](https://github.com/dart-lang/site-webdev/commits/4-dev/examples/ng)
+<!-- * [...add other relevant PRs here...] -->
+
+
+### API doc changes
```
Comment (chalin) [inline src/angular/version.md]: I would just have this content merged with the text in Ng 4.0 (alpha)

File: src/angular/version.md [hunk 5]
```diff
@@ -117,3 +117,132 @@ which is documented at [angular.io](https://angular.io).
...
+
+* angular2.common &rarr; angular2 <br>
+  Example: .../angular2.common/NgFor-class &rarr; [.../angular2/NgFor-class](https://webdev.dartlang.org/angular/api/angular2/NgFor-class)
+* [The AngularDart API doc homepage](/angular/api)
+  now shows an import statement next to each library's heading.
+
+
+### Dart 1.24
+
+The examples changed to use the Dart development compiler (dartdevc)
```
Comment (chalin) [inline src/angular/version.md]: I'd say that we did _initial work to prepare_ the examples for use of the dartdevc (I don't think that we are officially recommending its use yet ... and our Travis jobs span the use of both compilers).

File: src/angular/version.md (hunk 1 above)
Comment (kwalrath) [inline src/angular/version.md]: You're right, we should autogenerate a TOC for this. (It'd be nice to have the option to include 3rd-level headings in the autogenerated TOC! I'll check whether there's a bug for that.)

I think I'll make this its own page, and just call it changelog.md.

Comment (chalin) [inline src/angular/version.md]: > make this its own page, and just call it changelog.md.

+1

Comment (kwalrath) [inline src/angular/version.md]: I've filed #868, for completeness' sake.

File: src/angular/version.md (hunk 4 above)
Comment (kwalrath) [inline src/angular/version.md]: The only issue is I wanted to be clear that this change is already visible in the 3.0 docs.

File: src/angular/version.md (hunk 2 above)
Comment (kwalrath) [inline src/angular/version.md]: I figure they can easily walk up the tree if they want to.

File: src/angular/version.md (hunk 4 above)
Comment (chalin) [inline src/angular/version.md]: Ah. Maybe include the versions in the title then?

**Comment by kwalrath** [comment]
I've updated the staging site:

https://kw-webdev-dartlang-1.firebaseapp.com/angular/changelog
https://kw-webdev-dartlang-1.firebaseapp.com/angular/version

NOTE: The API doc build isn't working for me right now, so links to the current API docs won't work (but they were good before my build went kablooey).

File: src/angular/version.md [hunk 6]
```diff
@@ -88,6 +88,13 @@ packages. These packages often have a more recent development _(dev)_ release.
   {% endif %}
 </table>
 
+<aside class="alert alert-info" markdown="1">
+**Migration tip:**
+To see how the AngularDart documentation and examples have changed
+in response to each release, read the [changelog](/angular/changelog).
+</aside>
```
Comment (chalin) [inline src/angular/version.md]: Very nice!

File: src/angular/changelog.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+  * [angular2 changelog](https://pub.dartlang.org/packages/angular2#pub-pkg-tab-changelog)
+  * [angular changelog](https://pub.dartlang.org/packages/angular/versions/4.0.0-alpha%2B1#pub-pkg-tab-changelog)
+  * [angular_components changelog](https://pub.dartlang.org/packages/angular_components#pub-pkg-tab-changelog)
+
+
+## AngularDart 4.0 (alpha)
+
+<aside class="alert alert-info" markdown="1">
+**Note:**
+Most of these changes aren't yet visible on webdev.dartlang.org,
```
Comment (chalin) [inline src/angular/changelog.md]: How about:
> **Note:** These changes aren't yet visible on webdev.dartlang.org, but you can see a preview at ...

File: src/angular/changelog.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+Others are the result of new recommendations or documentation.
+
+Also see:
+
+* [History of commits to AngularDart documentation](https://github.com/dart-lang/site-webdev/commits/master/src/angular)
+* [History of commits to AngularDart examples](https://github.com/dart-lang/site-webdev/commits/master/examples/ng/doc)
+* Package changelogs:
+  * [angular2 changelog](https://pub.dartlang.org/packages/angular2#pub-pkg-tab-changelog)
+  * [angular changelog](https://pub.dartlang.org/packages/angular/versions/4.0.0-alpha%2B1#pub-pkg-tab-changelog)
+  * [angular_components changelog](https://pub.dartlang.org/packages/angular_components#pub-pkg-tab-changelog)
```
Comment (chalin) [inline src/angular/changelog.md]: Nit. These look funny w/o capitalization. Maybe, e.g.:
> * [`angular` changelog](#)

This comment applies here and further below.

File: src/angular/changelog.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+More information:
+
+* [angular changelog](https://pub.dartlang.org/packages/angular/versions/4.0.0-alpha%2B1#pub-pkg-tab-changelog)
+* [History for site-webdev/examples/ng (4-dev branch)](https://github.com/dart-lang/site-webdev/commits/4-dev/examples/ng)
+* [4.0 prep tracking issue](https://github.com/dart-lang/site-webdev/issues/670)
+
+
+## API doc changes (July 2017)
+
+Because `common.dart` is going away in 4.0, we had to change API doc generation.
```
Comment (chalin) [inline src/angular/changelog.md]: we had to change --> we changed

File: src/angular/changelog.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+We also fixed some bugs in the API doc homepage
+and added import information for each library.
+
+* angular2.common &rarr; angular2 <br>
+  Example: .../angular2.common/NgFor-class &rarr;
+  [.../angular2/NgFor-class](/angular/api/angular2/NgFor-class)
+* [The AngularDart API doc homepage](/angular/api)
+  now shows an import statement next to each library's heading.
```
Comment (chalin) [inline src/angular/changelog.md]: Nit: import --> `import` (that will make it stand out)

File: src/angular/changelog.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+More information:
+
+* [angular2 changelog](https://pub.dartlang.org/packages/angular2#pub-pkg-tab-changelog)
+* [PR #625](https://github.com/dart-lang/site-webdev/pull/625/files?w=1)
+
+
+## Testing doc and code, part 1
+
+Created an initial [Testing page](/angular/guide/testing) and
+added component tests to the tutorial examples.
```
Comment (chalin) [inline src/angular/changelog.md]: Do we want to say in a side note that a major review of this page is in preparation?

Comment (kwalrath) [inline src/angular/changelog.md]: I'd prefer not to pre-announce.

File: src/angular/changelog.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+## AngularDart 3.1
+
+The examples changed to update the release and reflect new recommendations.
+
+* Changed `core.dart` imports to `angular2.dart`.
+* To prepare for 3.1's experimental compiler, in each example:
+  * Removed `platform_directives` from `pubspec.yaml`,
+    moving the directives it listed
+    (for example, `COMMON_DIRECTIVES`, `CORE_DIRECTIVES`, `FORM_DIRECTIVES`, `ROUTER_DIRECTIVES`)
+    to the relevant components' `directives` lists.
```
Comment (chalin) [inline src/angular/changelog.md]: As @Janamou [pointed out](https://github.com/dart-lang/site-webdev/issues/856#issuecomment-317730150), we should mention that these now need to be added to app components, as appropriate &mdash; then it is probably best to point readers to the examples ... or the PR in which those changes was made.

Comment (kwalrath) [inline src/angular/changelog.md]: #625 does that, but I'll clarify.

Comment (kwalrath) [inline src/angular/changelog.md]: In fact, I decided to just show an example.

**Comment by kwalrath** [comment]
PTAL. @Janamou, does this work for you?

https://kw-webdev-dartlang-1.firebaseapp.com/angular/changelog
https://kw-webdev-dartlang-1.firebaseapp.com/angular/version

**Comment by kwalrath** [comment]
I'm going to publish this, and we can iterate as we get feedback.

**Pull request merged**